recipes/icon/build.sh

```diff
@@ -0,0 +1,133 @@
+mkdir -p ${DEPLOY}/ipl/cfuncs; cp    ${SRC_DIR}/ipl/cfuncs/* ${DEPLOY}/ipl/cfuncs/
+mkdir -p ${DEPLOY}/ipl/data  ; cp    ${SRC_DIR}/ipl/data/*   ${DEPLOY}/ipl/data/
+mkdir -p ${DEPLOY}/ipl/docs  ; cp    ${SRC_DIR}/ipl/docs/*   ${DEPLOY}/ipl/docs/
+mkdir -p ${DEPLOY}/ipl/incl  ; cp    ${SRC_DIR}/ipl/incl/*   ${DEPLOY}/ipl/incl/
+mkdir -p ${DEPLOY}/ipl/packs ; cp -R ${SRC_DIR}/ipl/packs/*  ${DEPLOY}/ipl/packs
+mkdir -p ${DEPLOY}/ipl/procs ; cp    ${SRC_DIR}/ipl/procs/*  ${DEPLOY}/ipl/procs/
+mkdir -p ${DEPLOY}/ipl/progs ; cp    ${SRC_DIR}/ipl/progs/*  ${DEPLOY}/ipl/progs/
+
+# Save cfuncs.u? because we have to omit from the build the build tools
+#   needed to reconstruct it via ipl/cfuncs/Makefile
+cp ${DEPLOY}/lib/cfunc.u? ${DEPLOY}/
+# Removing the ucode from lib would reduce the size of the conda package by
+#   almost half; the cost of this would be a (slightly) prolonged first activation.
+du -sh --apparent-size ${DEPLOY}/lib > ${DEPLOY}/lib/initial_size
+rm ${DEPLOY}/lib/*.u?
+# However, it apparently is better to increase the size of the package rather
+#   than defering this task to the activation; the reason for removing lib/*.u?
+#   above is that the default *.u? includes some files (2 Mb out of 4.6 Mb)
+#   that are not relevant to this conda pkg.
+pushd ${DEPLOY}/ipl/procs
+unset LPATH
+../../bin/icont -usc *.icn; mv *.u? ../../lib
+popd
+du -sh --apparent-size ${DEPLOY}/lib > ${DEPLOY}/lib/final_size
+
+# restore cfuncs.u?
+mv ${DEPLOY}/cfunc.u? ${DEPLOY}/lib/
+
+# Create instructions to make extension libraries in C
+mkdir -p ${DEPLOY}/doc ; cp ${SRC_DIR}/doc/* ${DEPLOY}/doc/
+echo '
+#!/bin/env bash
+
+# To build loadable C functions as described at
+#   https://www.cs.arizona.edu/icon/uguide/cfuncs.htm
+# you will need to add the C compiler and some tools to your environment, e.g.:
+
+    # get the build tools
+    conda install -c conda-forge gcc_linux-64 make sed binutils
+
+    # set the envars that would be set by `conda activate ...`
+    . $CONDA_PREFIX/etc/conda/activate.d/activate-gcc_linux-64.sh
+    . $CONDA_PREFIX/etc/conda/activate.d/activate-binutils_linux-64.sh
+
+    # change to the directory with the example
+    set -x
+    pushd ${CONDA_PREFIX}/icon/ipl/packs/loadfunc
+
+    # build and verify the example
+    make CC=${CC} && ls -l libdemo.so && ./ddtest
+
+    # uncomment to clean up, if desired
+    #make clean
+
+    # return to previous directory
+    popd
+
+# Bug: Building with C++, e.g.,
+#   ${CONDA_PREFIX}/icon/ipl/packs/loadfuncpp
+# has not yet been successful in a conda environment.
+# It will be necessary but not sufficient to install gxx_linux-64.
+' >> ${DEPLOY}/doc/condagcc.txt
+
+# Create or append activation script to set envars and README
+mkdir -p ${PREFIX}/etc/conda/activate.d
+echo '#!/bin/sh
+# set up IPL envar
+export IPL_OLD=${IPL}
+export IPL=${CONDA_PREFIX}/icon/ipl
+if [ ! -e ${CONDA_PREFIX}/README_icon ]; then
+  cat > ${CONDA_PREFIX}/README_icon << .
```

Comment:
Can you do this when the package is being made, so that we don't make files in the activate scripts?

Reply:
@isuruf I don't see how to have the files show the actual paths.  To be clear, you are not concerned about managing the IPL envar, right?  If it should not (as described in the online help for conda) be managed through an activation script, then how can it be otherwise managed?

Reply:
> To be clear, you are not concerned about managing the IPL envar, right?

Yes, I'm not concerned about that.

> I don't see how to have the files show the actual paths.

When conda-build finds a text file with ${PREFIX} in it, it will mark that file and conda will replace that ${PREFIX} with what is in the user's computer at install time.